src/core/algorithms/gfd/egfd_validation.cpp

```diff
@@ -0,0 +1,915 @@
+#include "egfd_validation.h"
+
+#include <iostream>
+
+#include <boost/graph/vf2_sub_graph_iso.hpp>
+
+#include "config/equal_nulls/option.h"
+#include "config/names_and_descriptions.h"
+#include "config/option_using.h"
+#include "config/tabular_data/input_table/option.h"
+
+namespace algos {
+
+void EGfdValidation::CfDecompose(const graph_t& graph, std::set<vertex_t>& core,
+                                 std::vector<std::set<vertex_t>>& forest) {
+    if (boost::num_vertices(graph) == (boost::num_edges(graph) + 1)) {
+        typename boost::graph_traits<graph_t>::vertex_iterator it, end;
+        for (boost::tie(it, end) = vertices(graph); it != end; ++it) {
+            core.insert(*it);
+        }
+        return;
+    }
+
+    typename boost::graph_traits<graph_t>::vertex_iterator it, end;

```

Comment:
This class (as well as other algorithm classes) contains a lot of huge methods. Please try to decompose them as much as possible. I'll leave some thoughts ahead.

Reply:
e.g. this `for` block looks extractable to a separate method.

Tip: use anonymous namespace to avoid bloating the class definition with utility methods, see HyFD/HyUCC for examples.